helm/values.yaml

```diff
@@ -72,17 +72,48 @@ minio:
       emptyDir: {}
     cache: |
       emptyDir: {}
-  autoscaler:
+
+  externalscaler:
     enabled: false
-    minReplicas: 1
-    maxReplicas: 10
-    metrics: |
-      - type: Resource
-        resource:
-          name: cpu
-          target:
-            averageValue: 150m
-            type: AverageValue
+    image: docker.pkg.github.com/cwm-keda-external-scaler/iamazeem/cwm-keda-external-scaler
+    imagePullPolicy: IfNotPresent
+    resources:
+      requests:
+        cpu: "20m"
+        memory: "64Mi"
+      limits:
+        memory: "150Mi"
+    REDIS_HOST: "minio-logger"
+
+  scaledobject:
+    enabled: false
+    spec:
+      scaleTargetRef:
+        name: minio
+      pollingInterval: 10
+      cooldownPeriod:  60
+      minReplicaCount: 1
+      maxReplicaCount: 10
+      # advanced:
+      #   restoreToOriginalReplicaCount: true
+      #   horizontalPodAutoscalerConfig:
+      #     behavior:
+      #       scaleDown:
+      #         stabilizationWindowSeconds: 30
+      #         policies:
+      #         - type: Percent
+      #           value: 80
+      #           periodSeconds: 15
+      triggers:
+        - type: external
+          metadata:
+            scalerAddress: minio-external-scaler.{{ .Release.Namespace }}:50051
```

Comment:
I don't think this will work, `{{ .Release.Namespace }}` is not available in the values.yaml, it is not needed in any case, you can just use `scalerAddress: minio-external-scaler:50051` - the current namesapce is used by default

Reply:
With `REDIS_HOST: minio-logger`, the minio's audit webhook is failing and the logger's fluentd instance is unable to locate the Redis server to connect with. Also, restarting `coredns` service doesn't help.

Here are the steps to reproduce this:

1. `minikube start --driver=docker --kubernetes-version=v1.16.14`
2. `eval $(minikube -p minikube docker-env)`
3. `export COMPOSE_TLS_VERSION=TLSv1_2`
4. `docker-compose build`
5. Configure and deploy with the scaler first i.e. `externalscaler.enabled=true`:

```yaml
minio:
  image: minio
  tag: latest
  initDebugEnable: true
  metricsLogger:
    image: cwm-worker-logger
    tag: latest
    LOG_LEVEL: debug
    LOG_PROVIDER: default
    UPDATE_GRACE_PERIOD_SECONDS: 2s
    DEPLOYMENT_API_METRICS_FLUSH_INTERVAL_SECONDS: 5s

  externalscaler:
    enabled: true

  scaledobject:
    enabled: false
```

6. Deploy: `helm upgrade -f .values.yaml --install cwm-worker-deployment-minio ./helm`
7. The above step should reproduce the issue. The same errors can be seen in the [CI workflow](https://github.com/CloudWebManage/cwm-worker-deployment-minio/pull/17/checks?check_run_id=2575852170#step:3:673). The minio pod keeps failing at audit webhook. And, the logs from the logger pod will show that it keeps trying to connect with the Redis server.
8. If that works, the next will be to enable ScaledObject i.e. `scaledobject.enabled=true` and deploy it again.